packages/loader/container-loader/src/serializedStateManager.ts

```diff
@@ -131,6 +131,7 @@ export class SerializedStateManager {
 	private latestSnapshot: ISnapshotInfo | undefined;
 	private refreshSnapshotP: Promise<void> | undefined;
 	private readonly lastSavedOpSequenceNumber: number = 0;
+	private supportGetSnapshotApi: boolean | undefined;
```

Comment:
why does this need to be mutable? It seems like this degree of freedom isn't leveraged in container.ts.

Reply:
It doesn't need to, but its value is assigned on setInitialSnapshot or fetchSnapshot methods, not allowing me to use readonly. I could write a guard in a setter but I believe this pattern (of assigning once a variable on setInitialSnapshot or fetchSnasphot) will happen again with an eventual timer and other variables and it needs a better tactic so I suspect the setter approach will be removed soon.

Reply:
yeah... I can see that's what the PR does, but it's not clear to me why from the description or context I have.

By scoping the variable to the class lifetime but using it in async methods, you can get a lot of weird side behavior here. E.g. someone can call `fetchSnapshot(foo, false)`, then call it again with `fetchSnapshot(foo, true)` before the first call reaches the `refreshLatestSnapshot` block. I'm not sure if this is something we're explicitly trying to support, but based on usage in container.ts it seems like probably not because it doesn't seem like the config changes.

Reply:
Yeah we're not expecting that case to happens. Even in the case it does, I'll handle the case in the future as I said in the previous comment.

Reply:
why have a private member at all then? seems like you could just use the function parameter and leave fetchSnapshot unchanged

Reply:
I'm planning to have a separate method which will call refreshLatestSnapshot() periodically, which will be unrelated to current workflows.

Reply:
I don't want to block you here, so I'm approving, but:

1. could you add a comment on the variable declaration that it's explicitly not supposed to change over the lifetime of a container (besides being initialized), or perhaps even some runtime verification?
2. I don't have a ton of context in this area, but it feels like with some refactoring you could achieve a simpler setup. All that's needed to initialize this variable to the right value is a service definition and a config provider, and it seems like we could delay creating the function to fetch the snapshot until both of those are present and immediately decide which variant we're using for the rest of the container lifetime. This seems true even with the periodic refresh, as presumably that bit doesn't get kicked off until the container is in an attached state.
 
Please think about options in this realm as you make further changes.